projects/core/src/auth/user-auth/services/auth-http-header.service.ts

```diff
@@ -95,7 +97,18 @@ export class AuthHttpHeaderService {
     // Logout user
     // TODO(#9638): Use logout route when it will support passing redirect url
     this.authService.coreLogout();
+
+    // There might be 2 cases:
+    // 1. when user is already on some page (router is stable) and performs an UI action
+    // that triggers http call (i.e. button click to save data in backend)
+    // 2. when user is navigating to some page and a route guard triggers the http call
+    // (i.e. guard loading cms page data)
+    //
+    // In the second case, we want to remember the anticipated url before we navigate to
+    // the login page, so we can redirect back to that URL after user authenticates.
+    this.authRedirectService.saveCurrentNavigationUrl();
```

Comment:
I'm thinking if it is safe to do it between logout and redirect? Is there any reason for not doing that before logout? I'm thinking for a case, where some customer implementation would have `coreLogout` working synchronously and making a redirect there.